gno/p/role_manager/role_manager.gno

```diff
@@ -0,0 +1,187 @@
+package role_manager
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ownable"
+)
+
+const (
+	RoleUser  = "user"
+	RoleAdmin = "admin"
+)
+
+const (
+	PermissionAll       = "all"
+	PermissionWriteRole = "write:role"
+)
+
+type Role struct {
+	name        string
+	permissions *avl.Tree // permission -> struct{}
+	users       *avl.Tree // user addr -> struct{}
+}
+
+type RoleManager struct {
+	owner *ownable.Ownable
```

Comment:
I think we should allow to drop ownership, forcing the existence of a superadmin is against web3 promises

also I think the concept of owner is redundant with the concept of admin in that case and we can remove owner altogether

Reply:
An ownable allow the owner to drop or transfer ownership isn't ?
I think i have to add a cap to export the owner variable or create a getter to make it accessible and then the owner can use the properties of an ownable cf. [78120da](https://github.com/TERITORI/teritori-dapp/pull/1293/commits/78120da8e0ad3276d6024b3e28d1f42113de5700)

Reply:
yep it can, yep it needs to be exposed I think